### PR TITLE
fix(async-queries): make global async. queries cookie SameSite option configurable

### LINKED ISSUE #21183: Unable to define the 'SameSite' for Global Async Query async-token cookie
For the session cookie, the Superset configuration allows to override the default behaviour and configure the SameSite option through the `SESSION_COOKIE_SAMESITE` variable.
That option is not available for the Global Async Token JWT cookie. The default value `None` is then considered.

### Environment

(please complete the following information):

- superset version: 1.5.1 / 2.0.0

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by rdubois**
This PR fixes #21183

### ADDITIONAL INFORMATION

- [x] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Thanks for the fix, and sorry it hasn't gotten more attention. Trying to get it routed on a path to success!

File: superset/config.py [hunk 1]
```diff
@@ -1282,6 +1282,7 @@ def EMAIL_HEADER_MUTATOR(  # pylint: disable=invalid-name,unused-argument
 GLOBAL_ASYNC_QUERIES_REDIS_STREAM_LIMIT_FIREHOSE = 1000000
 GLOBAL_ASYNC_QUERIES_JWT_COOKIE_NAME = "async-token"
 GLOBAL_ASYNC_QUERIES_JWT_COOKIE_SECURE = False
+GLOBAL_ASYNC_QUERIES_JWT_COOKIE_SAMESITE = None  # One of [None, 'None', 'Lax', 'Strict']
```
Comment (villebro) [inline superset/config.py:1285]: nit: it's generally convention to add these as type annotations:
```suggestion
GLOBAL_ASYNC_QUERIES_JWT_COOKIE_SAMESITE: Optional[Literal['None', 'Lax', 'Strict']] = None
```

**Review by villebro — COMMENTED** [review]
Looks great! One minor nit, but happy to get this merged asap if you can update that type annotation!

**Review by villebro — APPROVED** [review]
LGTM, thanks for adding this option

**Pull request merged**